### PR TITLE
Use vendor tornado instead of system tornado

### PR DESCRIPTION
**Pull request by dwoz**
### Tests written?

Yes

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by waynew — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

File: tests/integration/__init__.py [hunk 1]
```diff
@@ -266,6 +266,7 @@ def start_zeromq_daemons(self):
                 ' * {LIGHT_YELLOW}Starting salt-master ... {ENDC}'.format(**self.colors)
             )
             sys.stdout.flush()
+            print("WTF 1")
```
Comment (Akm0d) [inline tests/integration/__init__.py]: Debugging statement, should be removed

File: tests/integration/__init__.py [hunk 2]
```diff
@@ -278,6 +279,7 @@ def start_zeromq_daemons(self):
                 fail_hard=True,
                 event_listener_config_dir=RUNTIME_VARS.TMP_CONF_DIR,
                 start_timeout=120)
+            print("WTF 2")
```
Comment (Akm0d) [inline tests/integration/__init__.py]: debugging statement should be removed

File: salt/utils/http.py [hunk 3]
```diff
@@ -70,12 +70,12 @@
 # pylint: enable=import-error,no-name-in-module
 
 # Don't need a try/except block, since Salt depends on tornado
-import tornado.httputil
-import tornado.simple_httpclient
-from tornado.httpclient import HTTPClient
+import salt.ext.tornado.httputil
+import salt.ext.tornado.simple_httpclient
+from salt.ext.tornado.httpclient import HTTPClient
```
Comment (Akm0d) [inline salt/utils/http.py]: Should `import salt.ext.tornado.httpclient` and `salt.ext.tornado.httpclient.HTTPClient` be used instead?

**Review by Akm0d — APPROVED** [review]

File: salt/utils/http.py (hunk 3 above)
Comment (dwoz) [reply salt/utils/http.py]: I'm not making any code cleanup in this PR to keep down on the amount of change. This should get fixed at some point in the future though.

File: tests/unit/test_vendor_tornado.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import salt.modules.cmdmod
+import salt.utils.platform
+
+log = logging.getLogger(__name__)
+
+
+@skipIf(not salt.utils.path.which('bash'), 'Bash needed for this test')
+class VendorTornadoTest(TestCase):
+    '''
+    Ensure we are no using any non vendor'ed tornado
```
Comment (garethgreenaway) [inline tests/unit/test_vendor_tornado.py]: Simple typo here:
`Ensure we are not using any non vendor'ed tornado`

File: salt/transport/zeromq.py [hunk 5]
```diff
@@ -1122,8 +1123,9 @@ def __init__(self, opts, addr, linger=0, io_loop=None):
         self.addr = addr
         self.linger = linger
         if io_loop is None:
-            install_zmq()
-            ZMQDefaultLoop.current()
+            #install_zmq()
```
Comment (garethgreenaway) [inline salt/transport/zeromq.py]: Leaving these as comments vs just removing them?

**Review by garethgreenaway — CHANGES_REQUESTED** [review]

File: tests/unit/test_vendor_tornado.py (hunk 4 above)
Comment (dwoz) [reply tests/unit/test_vendor_tornado.py]: Fixed

File: salt/transport/zeromq.py (hunk 5 above)
Comment (dwoz) [reply salt/transport/zeromq.py]: Fixed

**Review by garethgreenaway — APPROVED** [review]

File: salt/transport/client.py [hunk 6]
```diff
@@ -91,9 +91,10 @@ class AsyncChannel(object):
 
     @classmethod
     def _config_resolver(cls, num_threads=10):
-        from tornado.netutil import Resolver
-        Resolver.configure(
-                'tornado.netutil.ThreadedResolver',
+        #from salt.ext.tornado.netutil import Resolver
```
Comment (Ch3LL) [inline salt/transport/client.py]: remove comment?

File: salt/utils/zeromq.py [hunk 7]
```diff
@@ -5,8 +5,11 @@
 # Import Python libs
 from __future__ import absolute_import, print_function, unicode_literals
 
+import salt.ext.tornado
+#sys.modules['tornado'] = salt.ext.tornado
```
Comment (Ch3LL) [inline salt/utils/zeromq.py]: remove comment?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
also would be good to add note of this change in the changelog/release notes

**Review by Ch3LL — CHANGES_REQUESTED** [review]
can we get a note added to the changelog/release notes about this change?

**Comment by dwoz** [comment]
> can we get a note added to the changelog/release notes about this change?

Yes, I have one more pull request to make for this change too update the dependencies. I will add the release notes to the next PR.

**Review by Ch3LL — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**